pkg/reconciler/broker/resources/deployments.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+
+	"go.uber.org/multierr"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	appsv1listers "k8s.io/client-go/listers/apps/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+// UpdateVolumeGenerationForDeployment updates the volume generation annotation
+// for all pods in this deployment. This technique is used to trigger immediate
+// configmap volume refresh without restarting the pod, see
+// https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#mounted-configmaps-are-updated-automatically
+// NOTE: update spec.template.metadata.annotations on the deployment resource
+// causes pods to bre recreated.
+func UpdateVolumeGenerationForDeployment(kubeClient kubernetes.Interface, dl appsv1listers.DeploymentLister, pl corev1listers.PodLister, ns, name string) error {
+	d, err := dl.Deployments(ns).Get(name)
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			return fmt.Errorf("deployment %v/%v doesn't exist: %w", ns, name, err)
+		}
+		return fmt.Errorf("error getting deployment %v/%v: %w", ns, name, err)
+	}
+
+	selector, err := metav1.LabelSelectorAsSelector(d.Spec.Selector)
```

Comment:
Out of curiosity, it's us creating the deployments and we know the labels we put on the pods. Do we really need to get the label selector from the deployments?

Reply:
I am doing this so we don't have to make sure whatever we use here is consistent with the yamls, or in case the users change the yamls. 

Anyways, the idea is that once we start to implement BrokerCell, the the BrokerCell controller will provide the labels,  so yes we won't need to get the labels from the deployment.

Reply:
Add a TODO would be helpful.

Reply:
Done